wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/RecipeRetriever.java

```diff
@@ -10,36 +10,121 @@
  */
 package org.eclipse.che.api.workspace.server;
 
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
 import javax.inject.Inject;
+import javax.inject.Named;
+import javax.ws.rs.core.UriBuilder;
+import org.eclipse.che.api.core.model.workspace.config.Recipe;
+import org.eclipse.che.api.core.util.FileCleaner;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.commons.env.EnvironmentContext;
+import org.eclipse.che.commons.lang.IoUtil;
+import org.slf4j.Logger;
 
 /**
  * Handle how recipe is retrieved, either by downloading it with external location or by using the
  * provided content.
  *
- * @author Florent Benoit
+ * @author Alexander Garagatyi
  */
 public class RecipeRetriever {
+  private static final Logger LOG = getLogger(RecipeRetriever.class);
+
+  private final URI apiEndpoint;
+
+  @Inject
+  public RecipeRetriever(@Named("che.api") URI apiEndpoint) {
+    this.apiEndpoint = apiEndpoint;
+  }
+
+  /**
+   * Returns content of recipe either by getting it from provided {@link Recipe} object or by
+   * retrieving it from location set in provided recipe.
+   *
+   * @param recipe recipe of {@link org.eclipse.che.api.core.model.workspace.config.Environment}
+   * @return recipe content
+   * @throws NullPointerException when recipe is null
+   * @throws IllegalArgumentException when both recipe content and location are null or empty
+   */
+  public String getRecipe(Recipe recipe) throws InfrastructureException {
+    if (recipe.getContent() != null && !recipe.getContent().isEmpty()) {
```

Comment:
I think it is good practice to do arguments verification at the beginning of the method just like you did [here](https://github.com/eclipse/che/pull/6047/files#diff-f41f64d6c368433dd6712c257a23c6faR117)
@garagatyi wdyt?